libtenzir/builtins/operators/unroll.cpp

```diff
@@ -0,0 +1,217 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/collect.hpp>
+#include <tenzir/plugin.hpp>
+
+#include <caf/sum_type.hpp>
+
+namespace tenzir::plugins::unroll {
+
+namespace {
+
+auto unroll_type(const type& src, const offset& off, size_t index = 0) -> type {
+  TENZIR_ASSERT(index <= off.size());
+  if (index == off.size()) {
+    auto list = caf::get_if<list_type>(&src);
+    TENZIR_ASSERT(list);
+    return list->value_type();
+  }
+  auto record = caf::get_if<record_type>(&src);
+  TENZIR_ASSERT(record);
+  auto fields = std::vector<struct record_type::field_view>{};
+  auto current = size_t{0};
+  auto target = off[index];
+  for (auto&& field : record->fields()) {
+    if (current == target) {
+      fields.emplace_back(field.name, unroll_type(field.type, off, index + 1));
+    } else {
+      fields.push_back(field);
+    }
+    ++current;
+  }
+  return type{src.name(), record_type{fields}, collect(src.attributes())};
+};
+
+class unroller {
+public:
+  unroller(int64_t row, arrow::ArraySpan list, const offset& offset)
+    : row_{row}, list_{std::move(list)}, offset_{offset} {
+  }
+
+  void run(arrow::StructBuilder& builder, const arrow::StructArray& source) {
+    TENZIR_ASSERT(row_ < source.length());
+    process_struct(builder, source, 0);
+  }
+
+private:
+  void process_struct(arrow::StructBuilder& builder,
+                      const arrow::StructArray& source, size_t index) {
+    TENZIR_ASSERT(index < offset_.size());
+    for (auto i = 0; i < list_.length; ++i) {
+      auto result = builder.Append();
+      TENZIR_ASSERT(result.ok());
+    }
+    auto target = detail::narrow<int>(offset_[index]);
+    for (auto current = 0; current < builder.num_fields(); ++current) {
+      if (current == target) {
+        process(*builder.field_builder(target), *source.field(target),
+                index + 1);
+      } else {
+        auto outer = arrow::ArraySpan{*source.field(current)->data()};
+        outer.SetSlice(row_, 1);
+        for (auto i = 0; i < list_.length; ++i) {
+          auto fb = builder.field_builder(current);
+          auto result = fb->AppendArraySlice(outer, 0, 1);
+          TENZIR_ASSERT(result.ok());
```

Comment:
I've managed to run into this assertion with this pipeline:

```
❯ tenzir 'from https://storage.googleapis.com/tenzir-datasets/M57/suricata.json.zst read suricata --no-infer | where #schema == "suricata.dns" | unroll dns.answers'
```

<details>
<summary>Backtrace</summary>

```
[20:17:14.936] loaded configuration file: /Users/dominiklohmann/.config/tenzir/tenzir.yaml
[20:17:14.942] loaded plugin configuration file: /Users/dominiklohmann/.config/tenzir/plugin/s3.yaml
warning: json parser failed to find schema for 'suricata.ike' and skips the JSON object '{"timestamp":"2021-11-17T21:41:43.862300+0100","flow_id":2014704461047611,"pcap_cnt":8176,"event_type":"ike","src_ip":"184.105.139.88","src_port":19784,"dest_ip":"198.71.247.91","dest_port":500,"proto":"UDP","pkt_src":"wire/pcap","community_id":"1:iEvyz8Eoe1S9NdnoxJueDNTpS4M=","ike":{"version_major":1,"version_minor":0,"init_spi":"3e35c70729dfedef","resp_spi":"0000000000000000","message_id":0,"exchange_type":2,"payload":["SecurityAssociation","Transform","Proposal"],"ikev1":{"doi":1,"encrypted_payloads":false,"client":{"proposals":[{"alg_enc":"EncCastCbc","alg_enc_raw":6}]},"server":{}}}}'
warning: json parser failed to find schema for 'suricata.ike' and skips the JSON object '{"timestamp":"2021-11-18T06:58:43.590635+0100","flow_id":847910971294642,"pcap_cnt":22913,"event_type":"ike","src_ip":"184.105.139.68","src_port":54251,"dest_ip":"198.71.247.91","dest_port":500,"proto":"UDP","pkt_src":"wire/pcap","community_id":"1:Vk/cKTKAb/1fwE258p+poPslFzs=","ike":{"version_major":1,"version_minor":0,"init_spi":"3e35c70729dfedef","resp_spi":"0000000000000000","message_id":0,"exchange_type":2,"payload":["SecurityAssociation","Transform","Proposal"],"ikev1":{"doi":1,"encrypted_payloads":false,"client":{"proposals":[{"alg_enc":"EncCastCbc","alg_enc_raw":6}]},"server":{}}}}'
0x6d31000107acdcb0: (tenzir::detail::panic(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>)+0x20)
0xb47f000107acdfbc: (tenzir::detail::fail_assertion_impl(char const*, std::__1::basic_string_view<char, std::__1::char_traits<char>>, std::__1::source_location)+0x104)
0xa927800104b47054: (void caf::variant<caf::none_t, bool, long long, unsigned long long, double, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l>>, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l>>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::pattern, tenzir::ip, tenzir::subnet, unsigned char, std::__1::vector<tenzir::data, std::__1::allocator<tenzir::data>>, tenzir::detail::vector_map<tenzir::data, tenzir::data, std::__1::allocator<std::__1::pair<tenzir::data, tenzir::data>>, tenzir::detail::stable_map_policy>, tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>, std::__1::basic_string<std::byte, std::__1::char_traits<std::byte>, std::__1::allocator<std::byte>>>::apply_impl<void, caf::variant<caf::none_t, bool, long long, unsigned long long, double, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l>>, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l>>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::pattern, tenzir::ip, tenzir::subnet, unsigned char, std::__1::vector<tenzir::data, std::__1::allocator<tenzir::data>>, tenzir::detail::vector_map<tenzir::data, tenzir::data, std::__1::allocator<std::__1::pair<tenzir::data, tenzir::data>>, tenzir::detail::stable_map_policy>, tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>, std::__1::basic_string<std::byte, std::__1::char_traits<std::byte>, std::__1::allocator<std::byte>>>, caf::detail::variant_data_destructor&>(caf::variant<caf::none_t, bool, long long, unsigned long long, double, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l>>, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l>>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::pattern, tenzir::ip, tenzir::subnet, unsigned char, std::__1::vector<tenzir::data, std::__1::allocator<tenzir::data>>, tenzir::detail::vector_map<tenzir::data, tenzir::data, std::__1::allocator<std::__1::pair<tenzir::data, tenzir::data>>, tenzir::detail::stable_map_policy>, tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>, std::__1::basic_string<std::byte, std::__1::char_traits<std::byte>, std::__1::allocator<std::byte>>>&, caf::detail::variant_data_destructor&)+0x0)
0x601c000104f03b94: (tenzir::plugins::unroll::(anonymous namespace)::unroller::process_struct(arrow::StructBuilder&, arrow::StructArray const&, unsigned long)+0x388)
0xa69000104f03a6c: (tenzir::plugins::unroll::(anonymous namespace)::unroller::process_struct(arrow::StructBuilder&, arrow::StructArray const&, unsigned long)+0x260)
0x9d2f000104f04c78: (tenzir::plugins::unroll::(anonymous namespace)::unroll_operator::operator()(tenzir::generator<tenzir::table_slice>, tenzir::operator_control_plane&) const (.resume)+0x7ac)
0xaf77000107ce352c: (caf::detail::default_behavior_impl<std::__1::tuple<caf::typed_behavior<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)> tenzir::(anonymous namespace)::exec_node<tenzir::table_slice, tenzir::table_slice>(caf::stateful_actor<tenzir::(anonymous namespace)::exec_node_state<tenzir::table_slice, tenzir::table_slice>, caf::typed_event_based_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>>*, std::__1::unique_ptr<tenzir::operator_base, std::__1::default_delete<tenzir::operator_base>>, caf::typed_actor<caf::result<caf::message> (tenzir::atom::run, tenzir::invocation), caf::result<tenzir::rest_response> (tenzir::atom::proxy, tenzir::http_request_description), caf::result<void> (tenzir::atom::internal, caf::spawn_atom, tenzir::atom::plugin), caf::result<caf::actor> (caf::spawn_atom, tenzir::invocation), caf::result<caf::ok_atom> (caf::put_atom, caf::actor, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<caf::actor> (caf::get_atom, tenzir::atom::label, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::label, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (caf::get_atom, tenzir::atom::version), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (tenzir::atom::config), caf::result<caf::typed_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>> (caf::spawn_atom, tenzir::operator_box, tenzir::tag_variant<void, tenzir::table_slice, caf::intrusive_ptr<tenzir::chunk>>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int)>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int, bool)::'lambda'(tenzir::atom::internal, tenzir::atom::run), caf::typed_behavior<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)> tenzir::(anonymous namespace)::exec_node<tenzir::table_slice, tenzir::table_slice>(caf::stateful_actor<tenzir::(anonymous namespace)::exec_node_state<tenzir::table_slice, tenzir::table_slice>, caf::typed_event_based_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>>*, std::__1::unique_ptr<tenzir::operator_base, std::__1::default_delete<tenzir::operator_base>>, caf::typed_actor<caf::result<caf::message> (tenzir::atom::run, tenzir::invocation), caf::result<tenzir::rest_response> (tenzir::atom::proxy, tenzir::http_request_description), caf::result<void> (tenzir::atom::internal, caf::spawn_atom, tenzir::atom::plugin), caf::result<caf::actor> (caf::spawn_atom, tenzir::invocation), caf::result<caf::ok_atom> (caf::put_atom, caf::actor, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<caf::actor> (caf::get_atom, tenzir::atom::label, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::label, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (caf::get_atom, tenzir::atom::version), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (tenzir::atom::config), caf::result<caf::typed_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>> (caf::spawn_atom, tenzir::operator_box, tenzir::tag_variant<void, tenzir::table_slice, caf::intrusive_ptr<tenzir::chunk>>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int)>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int, bool)::'lambda'(tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>&), caf::typed_behavior<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)> tenzir::(anonymous namespace)::exec_node<tenzir::table_slice, tenzir::table_slice>(caf::stateful_actor<tenzir::(anonymous namespace)::exec_node_state<tenzir::table_slice, tenzir::table_slice>, caf::typed_event_based_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>>*, std::__1::unique_ptr<tenzir::operator_base, std::__1::default_delete<tenzir::operator_base>>, caf::typed_actor<caf::result<caf::message> (tenzir::atom::run, tenzir::invocation), caf::result<tenzir::rest_response> (tenzir::atom::proxy, tenzir::http_request_description), caf::result<void> (tenzir::atom::internal, caf::spawn_atom, tenzir::atom::plugin), caf::result<caf::actor> (caf::spawn_atom, tenzir::invocation), caf::result<caf::ok_atom> (caf::put_atom, caf::actor, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<caf::actor> (caf::get_atom, tenzir::atom::label, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::label, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (caf::get_atom, tenzir::atom::version), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (tenzir::atom::config), caf::result<caf::typed_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>> (caf::spawn_atom, tenzir::operator_box, tenzir::tag_variant<void, tenzir::table_slice, caf::intrusive_ptr<tenzir::chunk>>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int)>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int, bool)::'lambda'(tenzir::atom::pause), caf::typed_behavior<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)> tenzir::(anonymous namespace)::exec_node<tenzir::table_slice, tenzir::table_slice>(caf::stateful_actor<tenzir::(anonymous namespace)::exec_node_state<tenzir::table_slice, tenzir::table_slice>, caf::typed_event_based_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>>*, std::__1::unique_ptr<tenzir::operator_base, std::__1::default_delete<tenzir::operator_base>>, caf::typed_actor<caf::result<caf::message> (tenzir::atom::run, tenzir::invocation), caf::result<tenzir::rest_response> (tenzir::atom::proxy, tenzir::http_request_description), caf::result<void> (tenzir::atom::internal, caf::spawn_atom, tenzir::atom::plugin), caf::result<caf::actor> (caf::spawn_atom, tenzir::invocation), caf::result<caf::ok_atom> (caf::put_atom, caf::actor, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<caf::actor> (caf::get_atom, tenzir::atom::label, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::label, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (caf::get_atom, tenzir::atom::version), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (tenzir::atom::config), caf::result<caf::typed_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>> (caf::spawn_atom, tenzir::operator_box, tenzir::tag_variant<void, tenzir::table_slice, caf::intrusive_ptr<tenzir::chunk>>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int)>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int, bool)::'lambda'(tenzir::atom::resume), caf::typed_behavior<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)> tenzir::(anonymous namespace)::exec_node<tenzir::table_slice, tenzir::table_slice>(caf::stateful_actor<tenzir::(anonymous namespace)::exec_node_state<tenzir::table_slice, tenzir::table_slice>, caf::typed_event_based_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>>*, std::__1::unique_ptr<tenzir::operator_base, std::__1::default_delete<tenzir::operator_base>>, caf::typed_actor<caf::result<caf::message> (tenzir::atom::run, tenzir::invocation), caf::result<tenzir::rest_response> (tenzir::atom::proxy, tenzir::http_request_description), caf::result<void> (tenzir::atom::internal, caf::spawn_atom, tenzir::atom::plugin), caf::result<caf::actor> (caf::spawn_atom, tenzir::invocation), caf::result<caf::ok_atom> (caf::put_atom, caf::actor, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<caf::actor> (caf::get_atom, tenzir::atom::label, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::label, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (caf::get_atom, tenzir::atom::version), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (tenzir::atom::config), caf::result<caf::typed_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>> (caf::spawn_atom, tenzir::operator_box, tenzir::tag_variant<void, tenzir::table_slice, caf::intrusive_ptr<tenzir::chunk>>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int)>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int, bool)::'lambda'(tenzir::diagnostic&), caf::typed_behavior<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)> tenzir::(anonymous namespace)::exec_node<tenzir::table_slice, tenzir::table_slice>(caf::stateful_actor<tenzir::(anonymous namespace)::exec_node_state<tenzir::table_slice, tenzir::table_slice>, caf::typed_event_based_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>>*, std::__1::unique_ptr<tenzir::operator_base, std::__1::default_delete<tenzir::operator_base>>, caf::typed_actor<caf::result<caf::message> (tenzir::atom::run, tenzir::invocation), caf::result<tenzir::rest_response> (tenzir::atom::proxy, tenzir::http_request_description), caf::result<void> (tenzir::atom::internal, caf::spawn_atom, tenzir::atom::plugin), caf::result<caf::actor> (caf::spawn_atom, tenzir::invocation), caf::result<caf::ok_atom> (caf::put_atom, caf::actor, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<caf::actor> (caf::get_atom, tenzir::atom::label, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::label, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (caf::get_atom, tenzir::atom::version), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (tenzir::atom::config), caf::result<caf::typed_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>> (caf::spawn_atom, tenzir::operator_box, tenzir::tag_variant<void, tenzir::table_slice, caf::intrusive_ptr<tenzir::chunk>>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int)>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int, bool)::'lambda'(tenzir::atom::push, tenzir::table_slice&), caf::typed_behavior<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)> tenzir::(anonymous namespace)::exec_node<tenzir::table_slice, tenzir::table_slice>(caf::stateful_actor<tenzir::(anonymous namespace)::exec_node_state<tenzir::table_slice, tenzir::table_slice>, caf::typed_event_based_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>>*, std::__1::unique_ptr<tenzir::operator_base, std::__1::default_delete<tenzir::operator_base>>, caf::typed_actor<caf::result<caf::message> (tenzir::atom::run, tenzir::invocation), caf::result<tenzir::rest_response> (tenzir::atom::proxy, tenzir::http_request_description), caf::result<void> (tenzir::atom::internal, caf::spawn_atom, tenzir::atom::plugin), caf::result<caf::actor> (caf::spawn_atom, tenzir::invocation), caf::result<caf::ok_atom> (caf::put_atom, caf::actor, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<caf::actor> (caf::get_atom, tenzir::atom::label, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::label, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (caf::get_atom, tenzir::atom::version), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (tenzir::atom::config), caf::result<caf::typed_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>> (caf::spawn_atom, tenzir::operator_box, tenzir::tag_variant<void, tenzir::table_slice, caf::intrusive_ptr<tenzir::chunk>>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int)>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int, bool)::'lambda'(tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>&), caf::typed_behavior<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)> tenzir::(anonymous namespace)::exec_node<tenzir::table_slice, tenzir::table_slice>(caf::stateful_actor<tenzir::(anonymous namespace)::exec_node_state<tenzir::table_slice, tenzir::table_slice>, caf::typed_event_based_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>>*, std::__1::unique_ptr<tenzir::operator_base, std::__1::default_delete<tenzir::operator_base>>, caf::typed_actor<caf::result<caf::message> (tenzir::atom::run, tenzir::invocation), caf::result<tenzir::rest_response> (tenzir::atom::proxy, tenzir::http_request_description), caf::result<void> (tenzir::atom::internal, caf::spawn_atom, tenzir::atom::plugin), caf::result<caf::actor> (caf::spawn_atom, tenzir::invocation), caf::result<caf::ok_atom> (caf::put_atom, caf::actor, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<caf::actor> (caf::get_atom, tenzir::atom::label, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), caf::result<std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>> (caf::get_atom, tenzir::atom::label, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (caf::get_atom, tenzir::atom::version), caf::result<tenzir::detail::vector_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, tenzir::data>>, tenzir::detail::stable_map_policy>> (tenzir::atom::config), caf::result<caf::typed_actor<caf::result<void> (tenzir::atom::internal, tenzir::atom::run), caf::result<void> (tenzir::atom::start, std::__1::vector<caf::actor, std::__1::allocator<caf::actor>>), caf::result<void> (tenzir::atom::pause), caf::result<void> (tenzir::atom::resume), caf::result<void> (tenzir::diagnostic), caf::result<void> (tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>, unsigned long long), caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>> (caf::spawn_atom, tenzir::operator_box, tenzir::tag_variant<void, tenzir::table_slice, caf::intrusive_ptr<tenzir::chunk>>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int)>, caf::typed_actor<caf::result<void> (tenzir::diagnostic)>, caf::typed_actor<caf::result<void> (tenzir::metric)>, int, bool)::'lambda'(tenzir::atom::pull, caf::typed_actor<caf::result<void> (tenzir::atom::push, tenzir::table_slice), caf::result<void> (tenzir::atom::push, caf::intrusive_ptr<tenzir::chunk>)>&, unsigned long long)>, caf::detail::dummy_timeout_definition>::invoke(caf::detail::invoke_result_visitor&, caf::message&)+0x1140)
0xbc1f800106009de8: (caf::scheduled_actor::consume(caf::mailbox_element&)+0x2dc)
0xd20100010600a514: (caf::scheduled_actor::reactivate(caf::mailbox_element&)+0x20)
0x5c3400010600e574: (caf::scheduled_actor::resume(caf::execution_unit*, unsigned long)::$_3::operator()(caf::mailbox_element&) const+0x94)
0x411c000106006458: (caf::scheduled_actor::resume(caf::execution_unit*, unsigned long)+0x450)
0x915c800105f242ec: (caf::scheduler::worker<caf::policy::work_stealing>::run()+0x68)
0x1c13800105f24208: (void std::__1::__thread_execute[abi:ue170006]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, std::__1::thread caf::actor_system::launch_thread<caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'()>(char const*, caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'())::'lambda'(caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'()), caf::intrusive_ptr<caf::ref_counted>, 2ul>(std::__1::tuple<caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'(), std::__1::thread caf::actor_system::launch_thread<caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'()>(char const*, caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'())::'lambda'(caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'()), caf::intrusive_ptr<caf::ref_counted>>&, std::__1::__tuple_indices<2ul>)+0x6c)
0x1638800105f240c4: (void* std::__1::__thread_proxy[abi:ue170006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, std::__1::thread caf::actor_system::launch_thread<caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'()>(char const*, caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'())::'lambda'(caf::scheduler::worker<caf::policy::work_stealing>::start()::'lambda'()), caf::intrusive_ptr<caf::ref_counted>>>(void*)+0x34)
0x3e1380018a94e034: (_pthread_start+0x88)
[20:17:15.672] panic: assertion `result.ok()` failed @ /Users/dominiklohmann/Desktop/tenzir/unroll-operator/libtenzir/builtins/operators/unroll.cpp:72
[20:17:15.672] version: 4.11.2+g17a2c44ddc
[20:17:15.672] this is a bug, we would appreciate a report - thank you!
*** unhandled exception: [id: 14, name: exec-node, exception typeid: std.runtime_error]: assertion `result.ok()` failed @ /Users/dominiklohmann/Desktop/tenzir/unroll-operator/libtenzir/builtins/operators/unroll.cpp:72
error: assertion `result.ok()` failed @ /Users/dominiklohmann/Desktop/tenzir/unroll-operator/libtenzir/builtins/operators/unroll.cpp:72
 = note: std.runtime_error
 = note: pipeline execution failed
```

</details>

Reply:
Interesting, I got a `std::bad_cast` instead. Probably depends on the Arrow version?

Anyway, I fixed this. It was cause by incomplete support for extension types (yes, again). I also added two tests that cover the duplication of extension type values. Took much more time to implement than initially expected, but now it (hopefully) works. :pray: